src/TeamRegistry.sol

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+contract TeamRegistry {
+    // -------------------------------------------------------------------------
+    // Errors
+    // -------------------------------------------------------------------------
+
+    /// @notice Emitted when a member is already part of a team.
+    /// @param  member The member of the team.
+    error AlreadyInTeam(address member);
+
+    /// @notice Emitted when the caller is the team leader.
+    /// @param  teamId The ID of the team.
+    /// @param  leader The leader of the team.
+    error IsTeamLeader(uint32 teamId, address leader);
+
+    /// @notice Emitted when the user does not have a pending invite.
+    /// @param  teamId The ID of the team.
+    /// @param  member The member of the team.
+    error NoPendingInvite(uint32 teamId, address member);
+
+    /// @notice Emitted when the player is not a team member.
+    /// @param  teamId The ID of the team.
+    /// @param  player The player.
+    error NotInTeam(uint32 teamId, address player);
+
+    /// @notice Emitted when the caller is not the team leader.
+    /// @param  teamId The ID of the team.
+    /// @param  player The member of the team.
+    error NotTeamLeader(uint32 teamId, address player);
+
+    // -------------------------------------------------------------------------
+    // Events
+    // -------------------------------------------------------------------------
+
+    /// @notice Emitted when a new team is created.
+    /// @param  teamId The ID of the team.
+    /// @param  leader The leader of the team.
+    event TeamCreated(uint32 teamId, address leader);
+
+    /// @notice Emitted when a team invite has been accepted.
+    /// @param  teamId The ID of the team.
+    /// @param  member The member of the team that accepted the invite.
+    event TeamInviteAccepted(uint32 teamId, address member);
+
+    /// @notice Emitted when team leadership is transferred.
+    /// @param  oldLeader The old leader of the team.
+    /// @param  newLeader The new leader of the team.
+    event TeamLeadershipTransferred(uint32 teamId, address oldLeader, address newLeader);
+
+    // -------------------------------------------------------------------------
+    // Storage
+    // -------------------------------------------------------------------------
+
+    /// @notice Mapping for keeping track of which members are part of teams.
+    /// @dev Returns a uint8 that determines status of the team member.
+    ///      `0`: Neither in team nor invited
+    ///      `1`: Invited to team but not yet accepted
+    ///      `2`: Accepted invite and is team member
+    ///      `3`: Is team leader
+    mapping(uint32 => mapping(address => uint8)) public teamMemberStatus;
+
+    /// @notice Returns boolean if the address is a member of a team.
+    mapping(address => bool) public isTeamMember;
+
+    /// @notice Team ID counter. Starts at 1.
+    uint16 private teamId;
```

Comment:
should this be `uint32`?

Reply:
<img width="695" alt="Screen Shot 2023-11-15 at 2 48 43 PM" src="https://github.com/waterfall-mkt/curta/assets/66016924/a18b849b-7fe3-491a-8bc3-83af40ed876e">

Reply:
Will make `uint256`